src/wallet/wallet.cpp

```diff
@@ -4500,9 +4500,11 @@ util::Result<MigrationResult> MigrateLegacyToDescriptor(std::shared_ptr<CWallet>
     }
 
     // Make a backup of the DB
-    fs::path this_wallet_dir = fs::absolute(fs::PathFromString(local_wallet->GetDatabase().Filename())).parent_path();
-    fs::path backup_filename = fs::PathFromString(strprintf("%s_%d.legacy.bak", (wallet_name.empty() ? "default_wallet" : wallet_name), GetTime()));
-    fs::path backup_path = this_wallet_dir / backup_filename;
+    fs::path legacy_wallet_dir = fs::absolute(fs::PathFromString(local_wallet->GetDatabase().Filename())).parent_path();
+    // Use the wallet directory's name, rather than the wallet name, to avoid issues with wallet's whose names contain relative file paths.
```

Comment:
In commit "wallet: Fix relative path backup during migration." (db3974ee11b50c5cdccd277aa6d5cce660032718)

This comment doesn't seem right. A relative file path like "foo" would not cause issues here. The problems happen with paths that contains slashes, which is why the line below is calling `filename()` to drop everything before the last slash. 

Would suggest a different comment that explains what this is trying to do. Maybe: "Create a backup copy of wallet data inside the wallet directory with a unique and descriptive filename, based on the name of the wallet and the current timestamp. If this wallet is not the default wallet, the last component of the wallet's directory name is used as the backup filename since the full wallet name may contain slashes."

Reply:
> This comment doesn't seem right.

This was wrong, thanks! I've added a variation of the comment you suggested that I edited for brevity.